### ISSUE TITLE
Feature request: ability to serialize and deserialize state

### ISSUE THREAD

**Issue by brianchirls** [PAIR-code/umap-js#32]
Since fitting large data sets can take quite a long time, it would be useful to be able to save the state to disk or elsewhere in memory so computation can be resumed elsewhere.

A couple use cases:

1. Resume asynchronous fitting if process needs to abort before it's done.
2. Save results of fitting in a database so additional points can be transformed later, on other devices or in a parallel process (see #11).

Thanks

**Comment by brianchirls** [comment]
Extra bonus points if it can be serialized as an array buffer so the data can be transferred between workers rather than copied. But I realize that's a big ask.

**Comment by cannoneyed** [comment]
Yeah, I've thought about this a bit... I think it's worth doing. I'll try to prioritize this moving forward.

**Comment by alistairheath** [comment]
Was this ever solved or a solution suggested? I've tried a number of serialisation libraries with no luck.

**Comment by r03ert0** [comment]
did anyone manage to do the serialization/deserialization?
